src/WebJobs.Script.WebHost/Diagnostics/FunctionsSystemLogsEventSource.cs

```diff
@@ -70,12 +70,29 @@ public void LogFunctionMetricEvent(string SubscriptionId, string AppName, string
             }
         }
 
-        public void SetActivityId(string activityId)
+        public static ActivityIdScope SetActivityId(string activityId)
         {
-            Guid guid;
-            if (!string.IsNullOrEmpty(activityId) && Guid.TryParse(activityId, out guid))
+            Guid.TryParse(activityId, out Guid guid);
+
+            SetCurrentThreadActivityId(guid, out Guid oldId);
+
+            return new ActivityIdScope(oldId);
+        }
+
+        public struct ActivityIdScope : IDisposable
+        {
+            public static readonly ActivityIdScope Empty = default;
```

Comment:
Is this used anywhere?

Reply:
No. It was at first. Ended up keeping it, but I can remove